pkg/trace/agent/agent.go

```diff
@@ -349,6 +355,22 @@ func newChunksArray(chunks []*pb.TraceChunk) []*pb.TraceChunk {
 
 var _ api.StatsProcessor = (*Agent)(nil)
 
+// filterSpans removes all spans for which the provided FilterSpan function returns true
+func (a *Agent) filterSpans(p *api.Payload) {
```

Comment:
We should have a unit test for this individual function too, and ideally a benchmark too.

Reply:
Doesn't this function already have a unit test?